Set up basic AB test for circles

JustinPinner opened the pull request:
**Note:** the build is currently reported as failed but this is a teamcity <--> github problem and the build _did_ work on a different TC domain.

## What does this change?
Adds a new epic and banner test to partition the audience for circles in support

## What is the value of this and can you measure success?
Allows us to partition the audience for the test 

## Does this affect other platforms - Amp, Apps, etc?
There are implications for support frontend

## Screenshots
N/A

## Tested in CODE?
In progress...

PRBuilds commented:
PRbuilds results:


**Screenshots**
[wide.png](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19198/screenshots/books%252F2014%252Fmay%252F21%252Fguardian-journalists-jonathan-freedland-ghaith-abdul-ahad-win-orwell-prize-journalism/wide.png) • [desktop.png](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19198/screenshots/books%252F2014%252Fmay%252F21%252Fguardian-journalists-jonathan-freedland-ghaith-abdul-ahad-win-orwell-prize-journalism/desktop.png) • [tablet.png](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19198/screenshots/books%252F2014%252Fmay%252F21%252Fguardian-journalists-jonathan-freedland-ghaith-abdul-ahad-win-orwell-prize-journalism/tablet.png) • [mobile.png](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19198/screenshots/books%252F2014%252Fmay%252F21%252Fguardian-journalists-jonathan-freedland-ghaith-abdul-ahad-win-orwell-prize-journalism/mobile.png)



&#x1F49A; **Exceptions**
[thrown-exceptions.js](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19198/thrown-exceptions.js)





&#x1F49A; **A11y validation**
[a11y-report.txt](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19198/a11y-report.txt)



&#x1F49A; **Microdata Validation**
[microdata.txt](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19198/microdata.txt)

    

**Apache Benchmark Load Testing**
[loadtesting.txt](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19198/loadtesting.txt)



**LightHouse Reporting**
[lighthouse.html](https://s3-eu-west-1.amazonaws.com/prbuilds/PR-19198/lighthouse.html)


    
--automated message

JustinPinner commented on static/src/javascripts/projects/common/modules/experiments/acquisition-test-selector.js:
```diff
@@ -36,6 +37,7 @@ const isViewable = (v: Variant, t: ABTest): boolean => {
  * acquisition tests in priority order (highest to lowest)
  */
 export const acquisitionsTests: $ReadOnlyArray<AcquisitionsABTest> = [
+    supportEpicCircles, // TODO: check priority placement
```
Can someone please confirm that it's ok to pop this in at the head of this list? Thanks.

rupertbates replied:
Can't answer this definitively but it seems fine to me.

joelochlann replied:
Yeah this isn't a particularly straightforward call to make. In this case it should be below `acquisitionsEpicUSGunCampaign` since that overlaps with this test (both apply in the US) but is more specific (it applies to pages with a specific tag) - i.e. it displays on a subset of pageviews that this test will. As things stand, `supportEpicCircles` would end up overriding the epic on the US gun campaign pages.

We could do with making this safer/more explicit/more comprehensible

JustinPinner commented on static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js:
```diff
@@ -187,7 +187,7 @@ const makeABTestVariant = (
         ),
         contributeURL = addTrackingCodesToUrl({
-            base: contributionsBaseURL,
+            base: options.contributionsBaseURL || contributionsBaseURL,
```
While this isn't needed for this test, it seemed to make sense to also be able to do it for the contribute url too. Thoughts, anyone?

rupertbates replied:
Personally I think I might leave it out under the principle of YAGNI

JustinPinner replied:
You're probably right. I imagined myself in a position of doing something with this in the future and wondering why there's yet another inconsistency in the way things are done.

rupertbates commented on static/src/javascripts/projects/common/modules/experiments/tests/support-epic-circles.js:
```diff
@@ -0,0 +1,48 @@
+import { makeABTest } from 'common/modules/commercial/contributions-utilities';
+import { getSync as getGeoLocation } from 'lib/geolocation';
+
+const targetUrl = () => {
```
Is there any way to share this function between the two tests rather than copy/pasting?

JustinPinner replied:
Fair point. I think I wasn't sure if it'd work for both or whether one was geolocated and the other was edition-drive. Tests on CODE suggest it'd be ok to share though.

joelochlann commented on static/src/javascripts/projects/common/modules/experiments/tests/support-epic-circles.js:
```diff
@@ -0,0 +1,48 @@
+            id: 'variant',
+            products: [],
+            options: {
+                isUnlimited: true,
```
This would make the display frequency "always ask" for everyone, which we definitely don't want. You should leave this parameter out and then the epic will default to the standard "Ask four" policy

JustinPinner replied:
Ok, thanks. I'll remove that.

joelochlann commented on static/src/javascripts/projects/common/modules/experiments/tests/support-epic-circles.js:
```diff
@@ -0,0 +1,48 @@
+        return 'https://support.theguardian.com/us/contribute';
+    if (fromGeo === 'GB')
+        return 'https://support.theguardian.com/uk/contribute';
+    return 'https://support.theguardian.com/';
```
Very minor point, but maybe worth a comment to say that this condition shouldn't happen, given that the test is only UK & US - it slightly confused me when reading it

JustinPinner replied:
Agreed, it _shouldn't_ happen. I just wondered if `fromGeo` could ever fail, in which case it's safer to assume a default state.

JustinPinner commented:
Hi @joelochlann, I've (hopefully) addressed your comments, and @rupertbates' too. I'm just waiting for CODE to become available for re-testing, but if it generally looks ok to you, please would you 👍? 
Many thanks.
